File: omnivault-frontend/src/services/authService.js
import axiosInstance from "./axiosInstance";

const authService = {
  login: async (usernameOrEmail, password) => {
    const response = await axiosInstance.post("/auth/login", {
      usernameOrEmail,
      password,
    });
    return response.data;
  },

  register: async (userData) => {
    const response = await axiosInstance.post("/auth/register", userData);
    return response.data;
  },

  logout: async () => {
    const refreshToken = localStorage.getItem("refresh_token");
    await axiosInstance.post("/auth/logout", { refreshToken });

    // Clear all authentication-related local storage
    localStorage.removeItem("access_token");
    localStorage.removeItem("refresh_token");
    localStorage.removeItem("verification_email");
  },

  getCurrentUser: async () => {
    const response = await axiosInstance.get("/auth/user");
    return response.data;
  },

  // Profile Update Method
  updateProfile: async (userData) => {
    const response = await axiosInstance.put("/auth/profile", userData);
    return response.data;
  },

  changePassword: async (currentPassword, newPassword) => {
    const response = await axiosInstance.put("/auth/change-password", {
      currentPassword,
      newPassword,
    });
    return response.data;
  },
  // Delete Account Method
  deleteAccount: async (password) => {
    const response = await axiosInstance.post("/auth/delete-account", {
      password,
    });

    // Clear all local storage on successful deletion
    localStorage.removeItem("access_token");
    localStorage.removeItem("refresh_token");
    localStorage.removeItem("verification_email");

    return response.data;
  },

  // Email Verification Methods
  verifyEmail: async (token) => {
    const response = await axiosInstance.post(
      `/auth/verify/token?token=${token}`
    );
    return response.data;
  },

  verifyEmailWithOTP: async (email, otpCode) => {
    const response = await axiosInstance.post(
      `/auth/verify/otp?email=${email}&otpCode=${otpCode}`
    );
    return response.data;
  },

  resendVerificationEmail: async (email) => {
    const response = await axiosInstance.post(
      `/auth/resend-verification?email=${email}`
    );
    return response.data;
  },

  // Token Refresh Method
  refreshToken: async (refreshToken) => {
    const response = await axiosInstance.post("/auth/refresh", {
      refreshToken,
    });

    // Update tokens in local storage
    if (response.data.accessToken) {
      localStorage.setItem("access_token", response.data.accessToken);
    }
    if (response.data.refreshToken) {
      localStorage.setItem("refresh_token", response.data.refreshToken);
    }

    return response.data;
  },

  // Password Reset Methods (if applicable)
  requestPasswordReset: async (email) => {
    const response = await axiosInstance.post("/auth/reset-password/request", {
      email,
    });
    return response.data;
  },

  resetPassword: async (token, newPassword) => {
    const response = await axiosInstance.post("/auth/reset-password/confirm", {
      token,
      newPassword,
    });
    return response.data;
  },
};

export default authService;

File: omnivault-backend/src/main/java/com/omnivault/service/AuthService.java
package com.omnivault.service;

import com.omnivault.domain.dto.request.LoginRequest;
import com.omnivault.domain.dto.request.RegisterRequest;
import com.omnivault.domain.dto.request.TokenRefreshRequest;
import com.omnivault.domain.dto.response.AuthResponse;
import com.omnivault.domain.dto.response.UserDTO;
import com.omnivault.domain.model.User;

/**
 * Service interface for authentication operations
 */
public interface AuthService {

    /**
     * Register a new user
     *
     * @param registerRequest The registration details
     * @return Authentication response with tokens
     */
    AuthResponse register(RegisterRequest registerRequest);

    /**
     * Authenticate a user
     *
     * @param loginRequest The login credentials
     * @return Authentication response with tokens
     */
    AuthResponse login(LoginRequest loginRequest);

    /**
     * Refresh an access token using a refresh token
     *
     * @param refreshRequest The refresh token request
     * @return Authentication response with new tokens
     */
    AuthResponse refreshToken(TokenRefreshRequest refreshRequest);

    /**
     * Logout a user
     *
     * @param refreshToken The refresh token to invalidate
     */
    void logout(String refreshToken);

    /**
     * Get the currently authenticated user
     *
     * @return The authenticated user
     */
    User getCurrentUser();

    /**
     * Get the currently authenticated user as a DTO
     *
     * @return The authenticated user as a DTO
     */
    UserDTO getCurrentUserDTO();

    /**
     * Verify email using token
     *
     * @param token The verification token
     * @return True if verification successful
     */
    boolean verifyEmail(String token);

    /**
     * Verify email using OTP code
     *
     * @param email The user's email
     * @param otpCode The OTP code
     * @return True if verification successful
     */
    boolean verifyEmailWithOTP(String email, String otpCode);

    /**
     * Resend verification email
     *
     * @param email The user's email
     * @return True if email sent successfully
     */
    boolean resendVerificationEmail(String email);
}